nimble/controller/src/ble_ll_adv.c

```diff
@@ -2646,6 +2674,11 @@ ble_ll_adv_rx_req(uint8_t pdu_type, struct os_mbuf *rxpdu)
     }
 #endif
 
+    /* See if the device is already connected */
+    if (ble_ll_adv_already_connected(peer, peer_addr_type)) {
```

Comment:
as per spec you quoted, this should be only checked when we receive CONNECT_IND or AUX_CONNECT_REQ - we should still be able to scan peer we are connected to (unless spec does not allow for this)